server/athenian/api/align/queries/metrics.py

```diff
@@ -240,31 +240,28 @@ def _parse_requested_metrics(params: Mapping[str, Any]) -> list[_RequestedMetric
             )
             for p in raw_teams_params
         }
-        req_metrics.append(_RequestedMetric(name, metric_params, team_params))
+        req_metrics.append(TeamRequestedMetric(name, metric_params, team_params))
     return req_metrics
 
 
-class _RequestedMetric:
+class TeamRequestedMetric:
     """A metric requested, with optional params and params override per team."""
 
     def __init__(self, name: str, params: Optional[dict], teams_params: Optional[dict[int, dict]]):
+        """Init the TeamRequestedMetric."""
         self._name = name
         self._params = params or {}
         self._teams_params = teams_params or {}
 
     @property
     def name(self) -> str:
+        """Get the name of the metric."""
         return self._name
 
     def resolve_for_team(self, team_id: int) -> MetricWithParams:
-        resolved_params = self._params
-        try:
-            team_params = self._teams_params[team_id]
-        except KeyError:
-            pass
-        else:
-            resolved_params = resolved_params | team_params
-        return MetricWithParams(self._name, resolved_params)
+        """Resolve the requested metric for a specific team."""
```

Comment:
I realized merging was meaningless here, it's an override for team, not an extension